designs/2022-community-eslint-org/README.md

```diff
@@ -0,0 +1,328 @@
+- Repo: No ESLint repos, only the ones mentioned below (owned by private
+  persons)
+- Start Date: 2022-05-31
+- RFC PR: <https://github.com/eslint/rfcs/pull/91>
+- Authors: [@aladdin-add](https://github.com/aladdin-add),
+  [@MichaelDeBoey](https://github.com/MichaelDeBoey) &
+  [@ota-meshi](https://github.com/ota-meshi)
+
+# `@eslint-community` GitHub organization
+
+## Summary
+
+<!-- One-paragraph explanation of the feature. -->
+
+Facilitate a GitHub organization (eg.
+[@eslint-community](https://github.com/eslint-community)) where community
+members can help ensure widely depended upon ESLint related packages stay up to
+date with newer ESLint releases and doesn't hold the wider community back.
+
+## Motivation
+
+<!-- Why are we doing this? What use cases does it support? What is the expected
+outcome? -->
+
+There are some widely depended upon ESLint related packages that aren't kept up
+to date with eg. new ESLint releases. The community would like a place to
+collectively step in and ensure that these doesn't hold the wider community
+back.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+### How do you envision `@eslint-community`?
+
+Just like the [@jest-community](https://github.com/jest-community) org, the
+`@eslint-community` org would be a centralized home for ESLint related packages
+that don't belong in the main org or that are widely depended upon, aren't
+maintained anymore and are given back to the community.
+
+> Community repos for ESLint related projects
+
+The benefit of having the org in place would be that the community isn't
+dependent on one person's GitHub/npm account.  
+The added benefit is that we _could_ create a structure to streamline the
+packages more. We _could_ for instance decide things like using the same tools,
+supporting the same Node/ESLint versions, if we want to release all "rescued"
+packages under the `@eslint-community` npm scope or not, ...
+
+As a reference, `@jest-community` has the following goal & intention:
+
+> The goal of `@jest-community` is to have a loose but centralized home for
+> stuff `jest` core thinks is neat and worth having but doesn't belong in the
+> core itself. The org is a way to help ensure that these projects are not
+> dependent on one person's GitHub account in case they move on.
+
+### Which projects should be involved?
+
+I think all of [@aladdin-add](https://github.com/aladdin-add),
+[@ota-meshi](https://github.com/ota-meshi),
+[@voxpelli](https://github.com/voxpelli) & myself would be in for helping to
+maintain at least the following packages:
+[`@mysticatea/eslint-plugin`](https://github.com/mysticatea/eslint-plugin),
```

Comment:
I'm not familiar with `@mysticatea/eslint-plugin`, something more like the author's personal preference? I'd prefer only to accept "important" projects - do not accept the ones just because it has a high downloads.

Reply:
How do you measure importance if not with downloads?

Reply:
<img width="971" alt="Skärmavbild 2022-06-02 kl  11 08 19" src="https://user-images.githubusercontent.com/34457/171596464-0c12f54a-2a10-41d4-b651-cf0243652c7b.png">

That it has more than one public dependent module? Else its not critical to the ecosystem?

Here's `eslint-plugin-node` as a comparison, it has 2639 dependents:

<img width="936" alt="Skärmavbild 2022-06-02 kl  11 09 32" src="https://user-images.githubusercontent.com/34457/171596713-581bca97-b4ec-471d-bce9-440173a0f11e.png">

Reply:
`@mysticatea/eslint-plugin` depends as devDeps for most packages created by `@mysticatea` (but not compatible with ESLint v8). If we can take over the `@mysticatea` packages, we must first replace them with different eslint-plugin and eslint-config. If we clear that, I don't think we need to continue using `@mysticatea/eslint-plugin`.

Reply:
However, if we expect `@mysticatea` to come back to OSS and maintain the community packages together, it may make sense to keep the original eslint configuration.

Reply:
All of the other mentioned packages by @mysticatea are indeed dependent on `@mysticatea/eslint-plugin` in their `devDependencies`.
Since `@mysticatea/eslint-plugin` is not ESLInt 8 compatible (yet), a lot of tests of the other plugins are failing.

We could indeed remove the package from `devDependencies`, but I think we can keep maintaining the package as well.

Reply:
And we can probably leave any decision on that out of this RFC and focus on the guidelines to make such decisions instead.

Reply:
All of the other mentioned packages by @mysticatea are indeed dependent on `@mysticatea/eslint-plugin` in their `devDependencies`.
Since `@mysticatea/eslint-plugin` is not ESLInt 8 compatible (yet), a lot of tests of the other plugins are failing.

We could indeed remove the package from `devDependencies`, but I think we can keep maintaining the package as well.